test/bugle_forms/handlers/question_test.clj

```diff
@@ -0,0 +1,33 @@
+(ns bugle-forms.handlers.question-test
+  (:require
+   [bugle-forms.db.connection :as db]
+   [bugle-forms.factories :as factories]
+   [bugle-forms.fixtures :as fixtures]
+   [bugle-forms.handlers.question :as sut]
+   [bugle-forms.model.question :as question]
+   [bugle-forms.model.form :as form]
+   [bugle-forms.model.user :as user]
+   [clojure.test :refer [deftest is testing use-fixtures]]
+   [next.jdbc.sql :as sql]))
+
+(use-fixtures :once fixtures/setup-config fixtures/setup-db)
+(use-fixtures :each fixtures/clear-db)
+
+(deftest add-question
+  (testing "Add question handler successfully adds a question to form"
+    (let [user (factories/user)
+          {user-id :user-account/id} (user/insert! user)
+          form (factories/form {:form/owner user-id})
+          {form-id :form/id} (form/insert! form)
+          {question-text :text :as form-params}
+          (factories/question-form-params)
+          request {:form-params form-params
+                   :route-params {:form-id (str form-id)}
+                   :session {:user {:name (:user/name user)
+                                    :uuid user-id}}}
+          response (sut/add-question request)]
+      (is (= 303 (:status response)))
+      (is (.startsWith (get (:headers response) "Location") "/form-builder/"))
+      (is (not-empty (sql/find-by-keys db/datasource :question
```

Comment:
Tests for failure could be added

Reply:
The only failure states that I can think of for this handler is:
- Malformed parameters (responsibility of the validation middleware which has undergone testing)
- Some unexpected database error (responsibility of error handling middleware which has undergone testing)

Is there any other negative case I'm missing? Should I retest these things here as well?

Reply:
I discussed this with @prabhanshuguptagit since you were in a meeting. His opinion was that the two things do not need to be tested in the handler layer. The one benefit of re-testing database exceptions and malformed requests again in the domain functions is to verify if the correct middleware has been applied. Even then, that is better tested in the routing layer—I have opened a ticket for that.

Is there anything specific to the domain of "adding questions" that you feel is still missing? If not, does it look okay to merge?

Reply:
> Even then, that is better tested in the routing layer

I have done these. I'm merging now. Do let me know if any change is required.